### ISSUE TITLE
multi gpu transformer of v1 seems much slow than v2 

### ISSUE THREAD

**Issue by venuswu** [tensorflow/models#7500]
I ran transformer of v1 on V100, but multi-gpu seems much slow like follows:
GPU 			1 | 1.8 			step/s
-- | --
GPU 			8 | 0.4 			step/s
But multi-gpu of transformer of v2 performs nearly as fast as the single-gpu. 
It looks like v1 and v2  both use the same distribution strategy--- tensorflow.python.distribute.mirrored_strategy.MirroredStrategyV1 .

But for v1, the distribution strategy reports as follows:
batch_all_reduce:1 all-reduces with algorithm = nccl,num_packs = 1, agg_small_grads_max_bytes = 0 and agg_small_grads_max_group = 10

for v2, it reports as follows: 
batch_all_reduce:187 all-reduces with algorithm = nccl, num_packs = 1, agg_small_grads_max_bytes = 0 and agg_small_grads_max_group = 10

**Comment by venuswu** [comment]
@tfboyd @reedwm

**Comment by venuswu** [comment]
> If you are using TensorFlow 2.0 you do not want to mess with the "V1" of the model. I am not even running it to convergence any more. The "v2" of the model for TensorFlow V2 is supported and trains daily with nightly Tensorflow and with the major releases so we know it works and we have stats.

yes, it diverges ! @tfboyd

**Comment by venuswu** [comment]
I ran the latest code of transformer on tf1.11, It seems that the variables are all ResourceVariable, why, I don't think the tf.enable_resource_variables() is opened.

**Comment by venuswu** [comment]
I have seen that int the repo: https://github.com/tensorflow/models, commitid: 45b708d48e87f4b52929286877d28efac81bf573, the file official/transformer/v2/transformer_main.py has changed.
Replace the optimizer.LazyAdam with tf.keras.optimizers.Adam, Why, does this?  Does the lazyadam damage accuracy, is it faster than optimizer??
@saberkun
